src/main/java/org/broadinstitute/dsm/export/ExportToES.java

```diff
@@ -126,20 +127,38 @@ public static void checkWorkflowNamesAndExport(List<String> workFlowColumnNames,
         DDPInstance ddpInstance = DDPInstance.getDDPInstanceById(instanceId);
         for (ParticipantDataDto participantData: allParticipantData) {
             String data = participantData.getData();
-            if (data != null) {
-                JsonObject dataJsonObject = gson.fromJson(data, JsonObject.class);
-                for (Map.Entry<String, JsonElement> entry: dataJsonObject.entrySet()) {
-                    if (workFlowColumnNames.contains(entry.getKey())) {
-                        ElasticSearchUtil.writeWorkflow(ddpInstance, participantData.getDdpParticipantId(),
-                                entry.getKey(), entry.getValue().getAsString());
-                    }
+            if (data == null) {
+                continue;
+            }
+            JsonObject dataJsonObject = gson.fromJson(data, JsonObject.class);
+            for (Map.Entry<String, JsonElement> entry: dataJsonObject.entrySet()) {
+                if (!workFlowColumnNames.contains(entry.getKey())) {
+                    continue;
                 }
-                if (dataJsonObject.has(MEMBER_TYPE) && dataJsonObject.get(MEMBER_TYPE).getAsString().equals(SELF)
-                    && dataJsonObject.has(FAMILY_ID)) {
-                    ElasticSearchUtil.writeDsmRecord(ddpInstance, null,
-                            participantData.getDdpParticipantId(), ESObjectConstants.FAMILY_ID, dataJsonObject.get(FAMILY_ID).getAsString(), null);
+                if (participantData.getFieldTypeId().equals(RGP_PARTICIPANTS)) {
+                    if (!dataJsonObject.has(FamilyMemberConstants.COLLABORATOR_PARTICIPANT_ID)
+                            || !dataJsonObject.has(FamilyMemberConstants.FIRSTNAME) || !dataJsonObject.has(FamilyMemberConstants.LASTNAME)) {
+                        continue;
+                    }
+                    if (ParticipantDataUtil.hasProbandEmail(participantData.getDdpParticipantId(),
```

Comment:
At this point you have already fetched `data` from `ddp_participant_data` but in `ParticipantDataUtil.hasProbandEmail` method you are fetching it again, why not to just pass already fetched data to this method?